src/scheduler/job_info.c

```diff
@@ -5054,14 +5054,35 @@ static int cull_preemptible_jobs(resource_resv *job, void *arg)
 			}
 			break;
 		case INSUFFICIENT_RESOURCE:
-			for (index = 0; job->select->chunks[index] != NULL; index++)
-			{
-				for (req_scan = job->select->chunks[index]->req; req_scan != NULL; req_scan = req_scan->next)
+			/* special check for vnode and host resource because those resources
+			 * do not get into chunk level resources. So in such a case we 
+			 * compare the resource name with the chunk name
+			 */
+			if (inp->err->rdef == getallres(RES_VNODE)) {
+				resource_req *hreq = find_resource_req(inp->job->resreq, inp->err->rdef);
+				if (hreq == NULL)
+					return 0;
+				for (index = 0; job->execselect->chunks[index] != NULL; index++)
 				{
-					if (req_scan->def == inp->err->rdef) {
-						if (req_scan->type.is_non_consumable ||
-						    req_scan->amount > 0) {
+					resource_req *lreq = find_resource_req(job->execselect->chunks[index]->req, inp->err->rdef);
```

Comment:
just trying to understand the logic: if a job requests a vnode explicitly, then here we are checking if any other job has also requested that vnode? What if a lowp job is just running on a vnode that's requested by the highp job? Shouldn't we also preempt such a job? (similar to how you handle the host case below)

Reply:
When resource requested by high priority job is a vnode then we check the execvnode list of running jobs and matches vnode name. This is because whether or not low priority job has requested a specific vnode, the vnode it is running on makes it into its execvnode list.
If I understand correctly, your second question is the bug we are trying to fix.

Reply:
but here you are looking for the vnode resource in lowp's execselect, not execvnode, right? I'm not sure I understand how this will check the execvnode list.

Reply:
I'm sorry I meant execselect, not execvnode